prime-dotnet-webapi/Controllers/OrganizationsController.cs

```diff
@@ -110,6 +123,117 @@ public async Task<ActionResult<Organization>> CreateOrganization(CreateOrganizat
+                }
+            }
+            var result = await _organizationClaimService.OrganizationClaimExistsAsync(search);
+
+            return Ok(result);
+        }
+
+        // GET: api/Organizations/5/claims
+        /// <summary>
+        /// Find OrganizationClaim by Organization ID.
+        /// </summary>
+        [HttpGet("{organizationId}/claims", Name = nameof(GetOrganizationClaimByOrgId))]
+        [Authorize(Roles = Roles.ViewSite)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiResultResponse<OrganizationClaim>), StatusCodes.Status200OK)]
+        public async Task<ActionResult<OrganizationClaim>> GetOrganizationClaimByOrgId(int organizationId)
+        {
+            var claim = await _organizationClaimService.GetOrganizationClaimAsync(organizationId);
+            if (claim == null)
+            {
+                return NotFound("No claim by a SigningAuthority exists for given Organization.");
+            }
+            return Ok(claim);
+        }
+
+        // POST: api/Organizations/5/claims/1/approve
+        /// <summary>
+        /// Approve claim for an existing Organization.
+        /// </summary>
+        [HttpPost("{organizationId}/claims/{claimId}/approve", Name = nameof(ApproveOrganizationClaim))]
+        [Authorize(Roles = Roles.EditSite)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<IActionResult> ApproveOrganizationClaim(int organizationId, int claimId)
+        {
+            var orgClaim = await _organizationClaimService.GetOrganizationClaimAsync(organizationId);
+            if (orgClaim == null || (orgClaim.Id != claimId))
+            {
+                return NotFound("Cannot locate Claim for given Organization.");
+            }
+            var userId = HttpContext.User.GetPrimeUserId();
+            Admin admin = await _adminService.GetAdminAsync(userId);
+
+            if (!await _organizationService.SwitchSigningAuthorityAsync(orgClaim.OrganizationId, orgClaim.NewSigningAuthorityId))
+            {
+                // TODO: Properly communicate error
+                return BadRequest("Could not assign Organization to new SigningAuthority.");
+            }
+
+            await _emailService.SendOrgClaimApprovalNotificationAsync(orgClaim);
+
+            await _businessEventService.CreateOrganizationEventAsync(orgClaim.OrganizationId, orgClaim.NewSigningAuthorityId, $"Organization Claim (Site ID/PEC provided: {orgClaim.ProvidedSiteId}, Reason: {orgClaim.Details}) approved by {admin.LastName}, {admin.FirstName}");
```

Comment:
I don't know if you really need to fetch the Admin here to add their name to the log; the business event has a direct reference to the Admin who created the event:
![image](https://user-images.githubusercontent.com/39168456/125719231-5d043178-9fba-4541-a1b8-96d32055e7cd.png)

Reply:
OK, agreed ... when I read description of https://bcgovmoh.atlassian.net/browse/PRIME-1655, I interpreted it differently at the time